File: FactoryExample/src/factoryexample/ArticuloInformatico.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package factoryexample;


/**
 *
 * @author fabm_
 */
public class ArticuloInformatico extends Articulo{
    
    private boolean isExento = false;
    
    public ArticuloInformatico(int id, String nombre, TipoArticulo tipoArticulo, long precio, boolean isExento) {
        super(id, nombre, tipoArticulo, precio);
        this.isExento = isExento;
    }

    @Override
    public long calcularIva() {
        if(isExento || (this.getPrecio() == 0)){
            return 0;
        }else{
            long iva = (long) (0.107 * this.getPrecio());
            return iva;
        }
    }
    
}

File: FactoryTs/articuloinformatico.ts
import { Articulo } from "./articulo";
import { TipoArticulo } from "./tipoarticulo";

export class ArticuloInformatico extends Articulo{

    public isExento: boolean;

    constructor(idArticulo:number, 
                                    nombre:string, 
                                    tipoArticulo:TipoArticulo, 
                                    precio:number, 
                                    isExento: boolean){
        super(idArticulo,nombre,tipoArticulo,precio);
        this.isExento = isExento;
    }

    public calcularIva():number{
        if(this.isExento == true || (this.getPrecio() == 0)){
            console.log(0);
            return 0;
        }else{
            console.log(0.107*this.getPrecio());
            return 0.107 * this.getPrecio();
        }
    }
}